maze.pl

```diff
@@ -9,6 +9,8 @@
 % that the element is in the list.
 %
 
+mem(X,[X|L]).
+mem(X,[Y|L]) :- mem(X,L).
```

Comment:
Note that neither of these rules stop as soon as they find that a member of the list matches.